Assets/Scripts/Models/Buildable/Furniture.cs

```diff
@@ -1306,6 +1306,20 @@ public void SetRotation(float rotation)
         Rotation = rotation;
     }
 
+    public Tile[] GetAllTiles()
+    {
+        Tile[] tiles = new Tile[Height * Width];
+        for (int x = 0; x < Width; x++)
+        {
+            for (int y = 0; y < Height; y++)
+            {
+                tiles[x + (y * Width)] = World.Current.GetTileAt(Tile.X + x, Tile.Y + y, Tile.Z);
+            }
+        }
+
+        return tiles;
+    }
+
```

Comment:
in my GetFurnitureTiles ( https://github.com/TeamPorcupine/ProjectPorcupine/pull/1657/files ) I used IEnumerable and yield hoping there will be less allocation of small arrays.. but not actually sure it's better (don't know Unity's mono that well)
 ```
   public static IEnumerable<Tile> GetFurnitureTiles(Tile startTile, Furniture furniture)
    {
        for (int x_off = startTile.X; x_off < startTile.X + furniture.Width; x_off++)
        {
            for (int y_off = startTile.Y; y_off < startTile.Y + furniture.Height; y_off++)
            {
                yield return World.Current.GetTileAt(x_off, y_off, startTile.Z);
            }
        }
    }
```